### PR TITLE
Don't bind sockaddr into sockaddr_storage unsafely

### PR DESCRIPTION
**Pull request by Reflejo**
By binding sockaddr into a sockaddr_storage we are actually assigning random memory bytes into the sockaddr_storage structre since sockaddr_t is actually smaller than sockaddr_storage.

### CODE REVIEW

**Comment by liangjiapei** [comment]
Thanks for fixing this issue! I wonder when this fix will be released to a new patch of pod.

**Comment by liangjiapei** [comment]
Any follow up?

**Comment by Reflejo** [comment]
I need to test ipv6 support since the test I had for it is actually
returning ipv4 addresses now. I’ll do it shortly

**Comment by keith** [comment]
@Reflejo can you test this?

**Comment by liangjiapei** [comment]
Any follow up? Thanks!

**Comment by Reflejo** [comment]
Sorry I didn't have time to set up a ipv6 network to try it but feel free to run the tests on an ipv6 network and check if it passes.

**Comment by Reflejo** [comment]
Hey @syedhali. If you have some free cycles, I'd love for someone with an ipv6 enabled network to try this. I can try to take a look over the weekend but would be really helpful to be sure. The way the tests are written right now are not forcing ipv6 and so this might be just routed through the ipv4 fallback

**Comment by syedhali** [comment]
@Reflejo thanks for the reply, that makes sense. I think ipv4 is 99% of use cases here so I'm not even sure I can test ipv6 lol. This PR looks like a safe refactor of the unsafe byte operations so if ipv4 functionality is working as expected we'll just work off this branch in our fork until someone with a ipv6 network can test and merge this into the master repo. Thanks again!

**Comment by Reflejo** [comment]
@syedhali how is it working for you so far

**Comment by syedhali** [comment]
> @syedhali how is it working for you so far

It's been working great!

File: Makefile [hunk 1]
```diff
@@ -10,8 +10,7 @@ install-tvOS:
 	true
 
 install-lint:
-	brew remove swiftlint --force || true
-	brew install --force-bottle https://raw.githubusercontent.com/Homebrew/homebrew-core/a97c85994a3f714355a20511b4df3a546ae809cf/Formula/swiftlint.rb
+	brew install swiftlint || true
```
Comment (keith) [inline Makefile:13]: the intent of this previous was version locking, nbd

**Review by keith — APPROVED** [review]

**Pull request merged**